### DISCUSSION TITLE
Random ShortURLs and Random Keyword Length

### DISCUSSION THREAD

**Discussion by deeusus** [YOURLS/YOURLS#2902] [Plugins]
If I understand correctly, the "Random Keyword Length" field sets the length of the randomly generated keyword(maximum keyword length?).
But what happens if all possible combinations run out?

**Comment by ozh** [comment]
YOURLS will enter an infinite loop trying to find a free short URL and the page will eventually time out.

**Reply by deeusus** [reply]
Thank you for your quick response)
Is it possible to fix it somehow? For example, if there is no free keyword, add a random character or automatically increase the value of "Random Keyword Length" +1

**Reply by ozh** [reply]
It's certainly *possible* to "fix" it, although it would be highly inefficient (would need to generate all possible random combinations and keep track of the one used)

So, I for one prefer the simple way where you just define the number of chars and are certain to never max it out :) All default chars, 4 times, is 15 million URLs. 5 and it's almost a billion.
